File: kotlin/lib/src/main/kotlin/api/Autoconfig.kt
// this file is @generated
package com.svix.kotlin.api

import com.svix.kotlin.SvixHttpClient
import com.svix.kotlin.models.AutoConfigOut
import com.svix.kotlin.models.RotateSubscriptionIn2
import okhttp3.Headers

data class AutoconfigCreateOptions(val idempotencyKey: String? = null)

data class AutoconfigRotateOptions(val idempotencyKey: String? = null)

class Autoconfig(private val client: SvixHttpClient) {
    /** Create an AutoConfig subscription. */
    suspend fun create(
        appId: String,
        options: AutoconfigCreateOptions = AutoconfigCreateOptions(),
    ): AutoConfigOut {
        val url = client.newUrlBuilder().encodedPath("/api/v1/app/$appId/autoconfig")
        val headers = Headers.Builder()
        options.idempotencyKey?.let { headers.add("idempotency-key", it) }
        return client.executeRequest<Any, AutoConfigOut>(
            "POST",
            url.build(),
            headers = headers.build(),
        )
    }

    /** Rotate the auth token and signing secret for an AutoConfig subscription. */
    suspend fun rotate(
        appId: String,
        autoconfigId: String,
        rotateSubscriptionIn2: RotateSubscriptionIn2,
        options: AutoconfigRotateOptions = AutoconfigRotateOptions(),
    ): AutoConfigOut {
        val url =
            client.newUrlBuilder().encodedPath("/api/v1/app/$appId/autoconfig/$autoconfigId/rotate")
        val headers = Headers.Builder()
        options.idempotencyKey?.let { headers.add("idempotency-key", it) }

        return client.executeRequest<RotateSubscriptionIn2, AutoConfigOut>(
            "POST",
            url.build(),
            headers = headers.build(),
            reqBody = rotateSubscriptionIn2,
        )
    }
}

File: php/src/Api/Autoconfig.php
<?php

// this file is @generated
declare(strict_types=1);

namespace Svix\Api;

use Svix\Exception\ApiException;
use Svix\Models\AutoConfigOut;
use Svix\Models\RotateSubscriptionIn2;
use Svix\Request\SvixHttpClient;

class Autoconfig
{
    public function __construct(
        private readonly SvixHttpClient $client,
    ) {
    }

    /**
     * Create an AutoConfig subscription.
     *
     * @throws ApiException
     */
    public function create(
        string $appId,
        AutoconfigCreateOptions $options = new AutoconfigCreateOptions(),
    ): AutoConfigOut {
        $request = $this->client->newReq('POST', "/api/v1/app/{$appId}/autoconfig");
        $request->setHeaderParam('idempotency-key', $options->idempotencyKey);
        $res = $this->client->send($request);

        return AutoConfigOut::fromJson($res);
    }

    /**
     * Rotate the auth token and signing secret for an AutoConfig subscription.
     *
     * @throws ApiException
     */
    public function rotate(
        string $appId,
        string $autoconfigId,
        RotateSubscriptionIn2 $rotateSubscriptionIn2,
        AutoconfigRotateOptions $options = new AutoconfigRotateOptions(),
    ): AutoConfigOut {
        $request = $this->client->newReq('POST', "/api/v1/app/{$appId}/autoconfig/{$autoconfigId}/rotate");
        $request->setHeaderParam('idempotency-key', $options->idempotencyKey);
        $request->setBody(json_encode($rotateSubscriptionIn2));
        $res = $this->client->send($request);

        return AutoConfigOut::fromJson($res);
    }
}
